core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -365,12 +365,29 @@ class WorkspaceService(protected val userInfo: UserInfo,
     }
   }
 
+  private def deleteWorkspaceTransaction(workspaceName: WorkspaceName, workspaceContext: Workspace) = {
+    dataSource.inTransaction { dataAccess =>
+      for {
+        // Delete components of the workspace
+        _ <- dataAccess.submissionQuery.deleteFromDb(workspaceContext.workspaceIdAsUUID)
+        _ <- dataAccess.methodConfigurationQuery.deleteFromDb(workspaceContext.workspaceIdAsUUID)
+        _ <- dataAccess.entityQuery.deleteFromDb(workspaceContext.workspaceIdAsUUID)
+
+        // Delete the workspace
+        _ <- dataAccess.workspaceQuery.delete(workspaceName)
+
+        // Schedule bucket for deletion
+        _ <- dataAccess.pendingBucketDeletionQuery.save(PendingBucketDeletionRecord(workspaceContext.bucketName))
+      } yield ()
+    }
+  }
+
   private def deleteWorkspace(workspaceName: WorkspaceName, workspaceContext: Workspace): Future[PerRequestMessage] = {
     //Attempt to abort any running workflows so they don't write any more to the bucket.
     //Notice that we're kicking off Futures to do the aborts concurrently, but we never collect their results!
     //This is because there's nothing we can do if Cromwell fails, so we might as well move on and let the
     //ExecutionContext run the futures whenever
-    val deletionFuture: Future[Seq[WorkflowRecord]] =
+    val abortWorkflowFuture: Future[Seq[WorkflowRecord]] =
       requesterPaysSetupService.revokeAllUsersFromWorkspace(workspaceContext).flatMap { _ =>
```

Comment:
I know this is out of scope for this issue, but not sure why this needs to be in this method.  Consider pulling it out into its own line in the calling `for` comp.  (but 100% OK if you just want to leave it for now, this is not what this ticket is about, but if we're refactoring and cleaning up the delete flow, it's something we could do to improve overall quality)